### PR TITLE
[Premium Feature] run_sql

### LINKED ISSUE #77: Add `run_sql` as premium helper
Building on top of the CRUD operations, we can implement a `run_sql` interface.

```py
def run_sql(raw_sql: str):
    - create_query
    - run_query
    - get_results
    - delete_query
```

### PR DESCRIPTION
**Pull request by bh2smith**
Closes #77 

Adds run_sql method which creates a query, executes, gets results and optionally archives the query (by default).

# Test Plan

This can't be tested with any API key that I have. But the test is there (skipped) and pretty straightforward to follow. Given that all the other routes are tested, it should be fine.

cc @TheEdgeOfRage & @diegoximenes for additional review.

### CODE REVIEW

**Review by diegoximenes — APPROVED** [review]

**Comment by diegoximenes** [comment]
Approved but e2e-tests are failing though :thinking:

**Comment by bh2smith** [comment]
> Approved but e2e-tests are failing though :thinking: 

This is partially because I have no more credits to test with. And also because upload CSV requires premium subscription which I also don't have.

**Comment by diegoximenes** [comment]
> This is partially because I have no more credits to test with. And also because upload CSV requires premium subscription which I also don't have.

I will check internally how Dune can give you access to an API key related to a premium subscription :)

File: dune_client/api/extensions.py [hunk 1]
```diff
@@ -123,6 +125,29 @@ def upload_csv(self, table_name: str, data: str, description: str = "") -> bool:
         except KeyError as err:
             raise DuneError(response_json, "UploadCsvResponse", err) from err
 
+    ##################
+    # Premium Features
+    ##################
```
Comment (msf) [inline dune_client/api/extensions.py:130]: add "these features use APIs that are only available on paid subscription plans"

**Review by msf — APPROVED** [review]

**Pull request merged**